src/java/org/apache/cassandra/service/reads/range/RangeCommands.java

```diff
@@ -72,19 +72,30 @@ static RangeCommandIterator rangeCommandIterator(PartitionRangeReadCommand comma
         Keyspace keyspace = Keyspace.open(command.metadata().keyspace);
         ReplicaPlanIterator replicaPlans = new ReplicaPlanIterator(command.dataRange().keyRange(), command.indexQueryPlan(), keyspace, consistencyLevel);
 
-        // our estimate of how many result rows there will be per-range
-        float resultsPerRange = estimateResultsPerRange(command, keyspace);
-        // underestimate how many rows we will get per-range in order to increase the likelihood that we'll
-        // fetch enough rows in the first round
-        resultsPerRange -= resultsPerRange * CONCURRENT_SUBREQUESTS_MARGIN;
         int maxConcurrencyFactor = Math.min(replicaPlans.size(), MAX_CONCURRENT_RANGE_REQUESTS);
-        int concurrencyFactor = resultsPerRange == 0.0
+        int concurrencyFactor = maxConcurrencyFactor;
+        Index.QueryPlan queryPlan = command.indexQueryPlan();
+        if ( queryPlan == null || queryPlan.shouldEstimateInitialConcurrency())
```

Comment:
The idea here is that normal queries and non-SAI indexes follow the current methodology of calculating a concurrency that is less than the maximum concurrency. For SAI queries `shouldEstimateInitialConcurrency == true` we use the maximum concurrency (generally the number of active endpoints). 

This is why we need the `MultiRangeReadCommand` because of the overhead of doing this.